### PR TITLE
stand up multiple agents (single and multi) for local development and testing

### LINKED ISSUE #2231: BUG: Mediation, Multitenant Aca-Py - sub-wallet initiates mediation
Version: 0.8.1

Scenario: Aca-Py in multitenant mode, using mediator that is initiated and managed by the tenant (type 2 - see [multitenancy / mediation](https://github.com/hyperledger/aries-cloudagent-python/blob/main/Multitenancy.md#mediation)).

Issue: Mediation can be granted, but when the tenant attempts to use mediation, the expected routing for the verkey cannot be found. For example, if a tenant creates an invitation using their mediation_id, when the invitation is received the verkey cannot be found and the connection cannot be completed.

```
Traceback (most recent call last):
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/transport/pack_format.py", line 178, in pack
    message = await wallet.pack_message(
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/wallet/askar.py", line 679, in pack_message
    raise WalletNotFoundError("Missing key for pack operation")
aries_cloudagent.wallet.error.WalletNotFoundError: Missing key for pack operation
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 256, in __step
    result = coro.send(None)
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/transport/pack_format.py", line 182, in pack
    raise WireFormatEncodeError("Message pack failed") from e
aries_cloudagent.transport.error.WireFormatEncodeError: Message pack failed
2023-05-12 17:49:06,048 aries_cloudagent.transport.outbound.manager ERROR Outbound message could not be delivered to https://1abe-207-6-152-178.ngrok.io
Traceback (most recent call last):
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/transport/pack_format.py", line 178, in pack
    message = await wallet.pack_message(
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/wallet/askar.py", line 679, in pack_message
    raise WalletNotFoundError("Missing key for pack operation")
aries_cloudagent.wallet.error.WalletNotFoundError: Missing key for pack operation
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 258, in __step
    result = coro.throw(exc)
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/transport/outbound/manager.py", line 483, in perform_encode
    queued.payload = await wire_format.encode_message(
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/transport/pack_format.py", line 156, in encode_message
    message = await (self.task_queue and self.task_queue.run(pack) or pack)
  File "/usr/local/lib/python3.9/asyncio/futures.py", line 284, in __await__
    yield self  # This tells Task to wait for completion.
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 328, in __wakeup
    future.result()
  File "/usr/local/lib/python3.9/asyncio/futures.py", line 201, in result
    raise self._exception
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 256, in __step
    result = coro.send(None)
  File "/home/indy/.local/lib/python3.9/site-packages/aries_cloudagent/transport/pack_format.py", line 182, in pack
    raise WireFormatEncodeError("Message pack failed") from e
aries_cloudagent.transport.error.WireFormatEncodeError: Message pack failed
```

If multitenant is using mediation of type 1 (default mediator set up with `--mediator-invitation` and `--mediator-connections-invite`) then mediation works. Mediation also works when agents are in single-tenant mode. This only happens with multitenancy and mediation type 2 configuration.

### PR DESCRIPTION
**Pull request by usingtechnology**
While working on load testing for [Aries Mediator Service](https://github.com/hyperledger/aries-mediator-service) I needed to spin up multiple agents and needed help to see a simple way to do so; at least not vanilla instances that I could run ad-hoc test scripts against.

I required multiple single-tenant mode and a multi-tenant mode to test our mediation between multiple agents. This was a simple way to achieve it. I decided I would stand up the standard `faber`, `alice`, `acme` single-tenant agents and one multi-tenant agent.

I did opt to include a Postgresql server shared across all the agents and included ngrok (easily turned off per agent) for URLs accessible outside the local network.

This was extremely helpful to diagnose and fix #2231

### CODE REVIEW

File: demo/playground/Dockerfile.acapy [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM ghcr.io/hyperledger/aries-cloudagent-python:py3.9-indy-1.16.0-0.8.1
```
Comment (swcurran) [inline demo/playground/Dockerfile.acapy:1]: Not sure about this so no worries if not changed, but wondered if you can use `../../docker/Dockerfile.demo` somehow.

File: demo/playground/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Aca-Py Playground
```
Comment (swcurran) [inline demo/playground/README.md:1]: Should be “ACA-Py” throughout.

File: demo/playground/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+This directory contains scripts to run several Aca-Py agents in various configurations for demonstration, development and testing scenarios. The agents are using Postgres (15) database storage (`askar`, `DatabasePerWallet`) and are running without security (`--admin-insecure-mode`).
+
+The inspiration for this playground was testing mediation and the differences between single-tenant and multi-tenant modes. These scripts allowed the developer to stand up 3 single-tenant agents and 1 multi-tenant agent and run various scenarios (in other scripts - not included). Running in `--admin-insecure-mode` simplifies creating tenants in multi-tenant mode and eliminates the need for adding headers for calls in single-tenant mode.
```
Comment (swcurran) [inline demo/playground/README.md:5]: Would it make sense to include one script as an example so it’s easier for others to be built?

Comment (usingtechnology) [reply demo/playground/README.md:5]: will do. i'll keep it self contained (ie. no mediator).

File: demo/playground/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+These configuration files are provided to the Aca-Py start command via the `AGENT_ARG_FILE` environment variable. See [`.env`](./.env.sample) and [`start.sh`](./start.sh).
+
+Additional (and more complicated) configurations coming soon... maybe.
+
+### Dockerfile and start.sh
+
+[`Dockerfile.acapy`](./Dockerfile.acapy) assembles the image to run. Currently based on [Aries Cloudagent Python 0.8.1](ghcr.io/hyperledger/aries-cloudagent-python:py3.9-indy-1.16.0-0.8.1), we need [jq](https://stedolan.github.io/jq/) to setup (or not) the ngrok tunnel and execute the Aca-py start command - see [`start.sh`](./start.sh). You may note that the start command is very sparse, additional configuration is done via environment variables in the [docker compose file](./docker-compose.yml).
+
+### ngrok
```
Comment (swcurran) [inline demo/playground/README.md:33]: Markdown Lint says to put a blank between an H* and the following text.

File: demo/playground/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+- to build the containers:
+
+```bash
+docker compose build
+```
+
+- to run the agents:
+
+```bash
+docker compose up
```
Comment (swcurran) [inline demo/playground/README.md:62]: On starting, I get some errors. Do I care?

```
playground-multi-agent-1        | 2023-05-12 20:36:45,292 aries_cloudagent.core.profile INFO Create profile manager: askar
playground-alice-agent-1        | 2023-05-12 20:36:45,299 aries_cloudagent.core.profile INFO Create profile manager: askar
playground-wallet-db-1          | 2023-05-12 20:36:45.314 UTC [80] FATAL:  database "multi_wallet" does not exist
playground-acme-agent-1         | 2023-05-12 20:36:45,320 aries_cloudagent.core.profile INFO Create profile manager: askar
playground-wallet-db-1          | 2023-05-12 20:36:45.323 UTC [81] FATAL:  database "alice_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.324 UTC [82] FATAL:  database "multi_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.333 UTC [83] FATAL:  database "alice_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.349 UTC [86] FATAL:  database "acme_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.360 UTC [87] FATAL:  database "acme_wallet" does not exist
```

**Review by swcurran — CHANGES_REQUESTED** [review]
A couple of tweaks - at least fix the CaPiTaLiZaTiOn of ACA-Py :-) .

I think it lacks a bit of “why do this?” motivation for how this can be used, so I think I would add how you are using it as an example, and include the stteps to run what you are running.  Not required if you don’t think it is worth it.

File: demo/playground/README.md (hunk 5 above)
Comment (usingtechnology) [reply demo/playground/README.md:62]: Not an issue, the databases will get created.

**Comment by usingtechnology** [comment]
> A couple of tweaks - at least fix the CaPiTaLiZaTiOn of ACA-Py :-) .
> 
> I think it lacks a bit of “why do this?” motivation for how this can be used, so I think I would add how you are using it as an example, and include the stteps to run what you are running. Not required if you don’t think it is worth it.

Added simple scripts, didn't want to include load testing and additional infrastructure etc as that gets out of date quickly.

**Review by swcurran — APPROVED** [review]

**Pull request merged**